### ISSUE TITLE
variables in TOML, Markdown, etc. formats?

### ISSUE THREAD

**Issue by kaicarver** [eobrain/bajel#17]
Hi,

The README says:

> the following examples [...] each of which specify the same thing

but the examples don't look the same, since the JavaScript ones use variables like  `CC`, `CFLAGS`, etc. 
You might want to make that clearer.

"if you want to use variables for complex builds":
I like things clean and readable, but makefile-style variables seem hard to do without, even for simple builds... 
So I tried doing a `build.mjs` but I get `Error: Not supported` and a stack trace ending with ` 'Error reading file': '/home/kai/fooproject/build.mjs'`

But anyway I bet that, for a project like this to work, it needs its own custom language.
Using a standard format for programming is tempting, but I suspect, a dead-end.
That's why we have so many programming languages, and why we don't program (much) in XML or whatever.

Nice idea for a project, anyway.

Though it seems quite new to be referenced in Wikipedia, which is how I came across it...
https://en.wikipedia.org/wiki/List_of_build_automation_software#Make-incompatible

**Comment by eobrain** [comment]
Merci!

I think you're probably right about variables being necessary for even simple build files, and despite wanting to keep things as simple as possible, I think this may be necessary. But I need to consider carefully what exact semantics the variables should have. I'll keep this Github issue open to track the feature request for variables.

I updated the README to incorporate some of your feedback.

The error you were seeing was probably because you were running a version of Node that does not support ES6 modules (.mjs files).  I just added some extra checking, and changed changed the error handling to replace the cryptic stack trace with a message of the form

```sh
ERROR: Need Node >=13.2.0 to use .mjs files (ES6 modules). Current version is v12.16.2.
```
I  published this change to npm as version 0.3.7 of bajel.
